includes/class-wc-calypso-bridge-woocommerce-admin.php

```diff
@@ -21,20 +21,37 @@ class WC_Calypso_Bridge_WooCommerce_Admin {
 	protected static $instance = false;
 
 	/**
-	 * Get class instance
+	 * Get class instance.
 	 */
 	public static function factory() {
 		if ( ! self::$instance ) {
 			self::$instance = new self();
 		}
+
 		return self::$instance;
 	}
 
 	/**
-	 * Initialize the class
+	 * Initialize the class.
 	 */
 	public function init() {
 		add_filter( 'wc_admin_get_feature_config', array( $this, 'maybe_remove_devdocs_menu_item' ) );
+
+		/**
+		 * Remove the legacy `WooCommerce > Coupons` menu.
+		 *
+		 * @since   1.9.4
+		 *
+		 * @param mixed $value Value to be filtered.
+		 *
+		 * @return int 1 to show the legacy menu, 0 to hide it. Booleans do not work.
+		 * @see     Automattic\WooCommerce\Internal\Admin\CouponsMovedTrait::display_legacy_menu()
+		 * @todo    Write a compatibility branch in CouponsMovedTrait to hide the legacy menu in new installations of WooCommerce.
+		 */
+		add_filter( 'pre_option_wc_admin_show_legacy_coupon_menu', function ( $value ) {
+			return 0;
+		}, PHP_INT_MAX );
+
```

Comment:
Hey @manospsyx - I've had a discussion with @xristos3490 and decided to use anonymous functions for these simple return functions. 

Anonymous functions were introduced in PHP 5.3, so we're good.

The fewer functions we introduce, the better. What do you think?

In other news, if `display_legacy_menu` supported booleans, it would be a lot easier  using `__return_false`

Reply:
@manospsyx approves :)